test/e2e/ffsvolume_creation.go

```diff
@@ -0,0 +1,87 @@
+	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/fss"
+	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/plugin"
+	"github.com/oracle/oci-volume-provisioner/test/e2e/framework"
+)
+
+var _ = Describe("FSS Volume Creation", func() {
+	f := framework.NewDefaultFramework("fss-volume")
+
+	It("Should be possible to create a persistent volume claim (PVC) for a FSS with a mnt target specified", func() {
+		pvcJig := framework.NewPVCTestJig(f.ClientSet, "volume-provisioner-e2e-tests-pvc")
+		By("Creating PVC that will dynamically provision a FSS")
+		pvc := pvcJig.CreateAndAwaitPVCOrFail(f.Namespace.Name, framework.VolumeFss, func(pvc *v1.PersistentVolumeClaim) {
+			pvc.Spec.Selector = &metav1.LabelSelector{MatchLabels: map[string]string{
+				plugin.LabelZoneFailureDomain: f.CheckEnvVar(framework.AD)}}
+
+			pvcJig.StorageClassName = framework.ClassOCIMntFss
+			pvcJig.CheckSCorCreate(pvcJig.StorageClassName, core.ProvisionerNameFss, map[string]string{
+				fss.MntTargetID: f.CheckEnvVar(framework.MntTargetOCID)})
+			pvc.Spec.StorageClassName = &pvcJig.StorageClassName
+
+		})
+		By("Creating a Pod and waiting till attaches to the volume")
+		pvcJig.CreateAndAwaitNginxPodOrFail(f.Namespace.Name, pvc)
+
+		pvcJig.DeleteStorageClass(pvcJig.StorageClassName)
+
+	})
+
+	It("Should be possible to create a persistent volume claim (PVC) for a FSS with a subnet id specified", func() {
+		pvcJig := framework.NewPVCTestJig(f.ClientSet, "volume-provisioner-e2e-tests-pvc")
+		By("Creating PVC that will dynamically provision a FSS")
+		pvc := pvcJig.CreateAndAwaitPVCOrFail(f.Namespace.Name, framework.VolumeFss, func(pvc *v1.PersistentVolumeClaim) {
+			pvc.Spec.Selector = &metav1.LabelSelector{MatchLabels: map[string]string{
+				plugin.LabelZoneFailureDomain: f.CheckEnvVar(framework.AD)}}
+
+			pvcJig.StorageClassName = framework.ClassOCISubnetFss
+			pvcJig.CheckSCorCreate(pvcJig.StorageClassName, core.ProvisionerNameFss, map[string]string{
+				fss.SubnetID: f.CheckEnvVar(framework.SubnetOCID)})
+			pvc.Spec.StorageClassName = &pvcJig.StorageClassName
+
+		})
+		By("Creating a Pod and waiting till attaches to the volume")
+		pvcJig.CreateAndAwaitNginxPodOrFail(f.Namespace.Name, pvc)
+
+		pvcJig.DeleteStorageClass(pvcJig.StorageClassName)
+
+	})
+
+	It("Should be possible to create a persistent volume claim (PVC) for a FSS no mnt target or subnet id specified", func() {
+		pvcJig := framework.NewPVCTestJig(f.ClientSet, "volume-provisioner-e2e-tests-pvc")
+		By("Creating PVC that will dynamically provision a FSS")
+		pvc := pvcJig.CreateAndAwaitPVCOrFail(f.Namespace.Name, framework.VolumeFss, func(pvc *v1.PersistentVolumeClaim) {
+			pvc.Spec.Selector = &metav1.LabelSelector{MatchLabels: map[string]string{
+				plugin.LabelZoneFailureDomain: f.CheckEnvVar(framework.AD)}}
+			pvcJig.StorageClassName = framework.ClassOCINoParamFss
+			pvcJig.CheckSCorCreate(pvcJig.StorageClassName, core.ProvisionerNameFss, nil)
```

Comment:
The tweak function is for customising the passed pvc only. This should be performed before the call to `CreateAndAwaitPVCOrFail`.